components/platform/include/components/platform/dl.hpp

```diff
@@ -0,0 +1,78 @@
+/* Copyright (c) 2022, Arm Limited and Contributors
```

Comment:
Are those files here really supposed to be at `components/platform/include/components/platform`, or just at `components/platform/include` ?

Reply:
Theres two layers to this.

`components/platform/` contains all files for the `platform` component. Everything in this folder is for the platform component and only the platform component.

`components/platform/include/components/platform` preserves the include path in source. `#include <components/platform/platform.hpp>`.

For now components are small so it looks a little overkill. However, when these components inevitably grow in size, this structure should scale with that growth.

I'm open to suggestions on how to improve this but I am hoping to merge the current open MRs before trying reorganizations of source files.

Reply:
The only real downside potential is the character limit in windows on file paths.  I'm fine with the path in this format as it makes the most sense to me.  However, we might consider renaming them to something else to avoid confusion.  Personally this is one I don't care about one way or the other.